### PR TITLE
docs: Update Android profiling documentation

### PR DESCRIPTION
**Pull request by ebariche**
## PR Type

What kind of change does this PR introduce?
- Documentation

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: doc/articles/guides/profiling-applications.md [hunk 1]
```diff
@@ -113,69 +113,58 @@ dotnet-trace collect --diagnostic-port ~/my-dev-port,connect --format speedscope
...
-<ItemGroup Condition="'$(AndroidEnableProfiler)'=='true'">
-    <AndroidEnvironment Condition="'$(IsEmulator)' == 'true'" Include="Platforms/Android/environment.emulator.txt" />
-    <AndroidEnvironment Condition="'$(IsEmulator)' != 'true'" Include="Platforms/Android/environment.device.txt" />
-</ItemGroup>
-```
-
-Then in the `Platforms/Android` application folder, add the following two files:
-
-- `environment.device.txt`
+- For devices
```
Comment (agneszitte) [inline doc/articles/guides/profiling-applications.md:122]:
```suggestion
- For devices:
```

File: doc/articles/guides/profiling-applications.md [hunk 2]
```diff
@@ -113,69 +113,58 @@ dotnet-trace collect --diagnostic-port ~/my-dev-port,connect --format speedscope
...
 
     ```text
-    DOTNET_DiagnosticPorts=127.0.0.1:9000,suspend
+    DOTNET_DiagnosticPorts=127.0.0.1:9000,suspend,connect
     ```
 
-- `environment.emulator.txt`
+- For emulators
```
Comment (agneszitte) [inline doc/articles/guides/profiling-applications.md:128]:
```suggestion
- For emulators:
```

File: doc/articles/guides/profiling-applications.md [hunk 3]
```diff
@@ -113,69 +113,58 @@ dotnet-trace collect --diagnostic-port ~/my-dev-port,connect --format speedscope
...
-    dotnet build -f net7.0-android -t:run -c Release -p:IsEmulator=true /p:RunAOTCompilation=true /p:AndroidEnableProfiler=true
+    dotnet build -c Release -f net8.0-android -r android-arm64 -t:Run -p:AndroidEnableProfiler=true
     ```
 
-- The app will start and the `dotnet-trace` will display a MB number counting up
-- Use the app and once done, stop `dotnet-trace` using the specified method (Likely `Enter` or `Ctr+C`)
-- Open a browser at `https://speedscope.app` and drop the `uno-app-trace.speedscope.json` file on it
+  Use `-r android-x64` for emulators instead.
+
+- The app will start and `dotnet-trace` will display a MB number counting up
```
Comment (agneszitte) [inline doc/articles/guides/profiling-applications.md:161]:
```suggestion
- The app will start and `dotnet-trace` will display an MB number counting up
```

**Review by jeromelaban — APPROVED** [review]

**Comment by agneszitte** [comment]
@ebariche additional changes coming for this PR?

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.2

**Pull request merged**